File: app/src/main/java/hr/kacan/trznica/adapters/MenuAdapter.java
package hr.kacan.trznica.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import com.squareup.picasso.Picasso;
import java.util.ArrayList;
import java.util.List;
import hr.kacan.trznica.R;
import hr.kacan.trznica.models.TipProizvoda;

public class MenuAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<TipProizvoda> items = new ArrayList<>();

    private Context ctx;
    private OnItemClickListener mOnItemClickListener;

    public interface OnItemClickListener {
        void onItemClick(View view, TipProizvoda tipProizvoda, int position);
    }

    public void setOnItemClickListener(final OnItemClickListener mItemClickListener) {
        this.mOnItemClickListener = mItemClickListener;
    }

    public MenuAdapter(Context context, List<TipProizvoda> items) {
        this.items = items;
        ctx = context;
    }

    public class OriginalViewHolder extends RecyclerView.ViewHolder {
        public ImageView image;
        public TextView title;
        public View lyt_parent;

        public OriginalViewHolder(View v) {
            super(v);
            image = (ImageView) v.findViewById(R.id.slika);
            title = (TextView) v.findViewById(R.id.naziv);
            lyt_parent = (View) v.findViewById(R.id.lyt_parent);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        RecyclerView.ViewHolder vh;
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.tip_card, parent, false);
        vh = new OriginalViewHolder(v);
        return vh;
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, final int position) {
        if (holder instanceof OriginalViewHolder) {
            OriginalViewHolder view = (OriginalViewHolder) holder;

            TipProizvoda tip = items.get(position);
            view.title.setText(tip.getNaziv());

            Picasso.get()
                    .load(tip.getSlika())
                    .fit()
                    .centerCrop()
                    .error(R.drawable.no_img)
                    .into(view.image);
            view.lyt_parent.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (mOnItemClickListener != null) {
                        mOnItemClickListener.onItemClick(view, items.get(position), position);
                    }
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }



}

File: app/src/main/java/hr/kacan/trznica/adapters/MenuAdapter.kt
package hr.kacan.trznica.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import hr.kacan.trznica.R
import hr.kacan.trznica.models.TipProizvoda
import java.util.*

class MenuAdapter(items: MutableList<TipProizvoda>, listener: OnItemClickListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var items: MutableList<TipProizvoda> = ArrayList()
    private var mOnItemClickListener: OnItemClickListener = listener

    interface OnItemClickListener {
        fun onItemClick(view: View, tipProizvoda: TipProizvoda, position: Int)
    }
    inner class OriginalViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var image: ImageView = v.findViewById<View>(R.id.slika) as ImageView
        var title: TextView = v.findViewById<View>(R.id.naziv) as TextView
        var lyt_parent: View = v.findViewById<View>(R.id.lyt_parent) as View
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val vh: RecyclerView.ViewHolder
        val v = LayoutInflater.from(parent.context).inflate(R.layout.tip_card, parent, false)
        vh = OriginalViewHolder(v)
        return vh
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is OriginalViewHolder) {
            val tip = items[position]
            holder.title.text = tip.naziv
            Picasso.get()
                    .load(tip.slika)
                    .fit()
                    .centerCrop()
                    .error(R.drawable.no_img)
                    .into(holder.image)

            holder.lyt_parent.setOnClickListener{ view ->
                mOnItemClickListener.onItemClick(view, items[position], position)
            }
        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    init {
        this.items = items

    }

}